WordPress/gutenberg
# Focal Point Picker

## Description
This PR creates a Focal Point Picker component and implements it in the `Cover` block. Here is the rationale for this functionality: when dealing with large background images like the ones used in `Cover`, it's common to see undesirable crops, especially on smaller viewports. If the primary subject of an image happens to be dead center then it will display perfectly everywhere, but if it is off-center by even a little, key visual information can be cropped out.

The Focal Point Picker component allows the user to visually select the ideal center point, which is then returned as a pair of percentage-like coordinates (e.g. `{ x: 0.1, y: 0.9 }`) which can easily be converted into `background-position` attributes and applied to the element with the background image. 

### Video and Parallax

The Focal Point Picker only works if an image is being used, and if `Fixed Background` is disabled. For now, the same approach won't be applied to videos, and Parallaxed images don't really require a focal point since their crop changes as the user scrolls.

### Legacy Content

As part of this branch, `Cover` block saves the dimensions of images when selected, which is needed by the component. For instances of `Cover` block that pre-date this branch there will be no dimensions data, and therefore the focal point picker is suppressed. Because of this approach there was no need to add a Gutenberg deprecation definition.

## How has this been tested?

- Spin up local Gutenberg instance following instructions here: https://github.com/WordPress/gutenberg/blob/master/CONTRIBUTING.md
- Switch branch to `try/focal-point-picker`
- Create new post
- Add `Cover` block to post.
- Upload or select a large image, ideally one where the central focus is not off center - the further the better.
- Click on block, then show block settings. In the sidebar, click and drag the picker icon to the 
focal point of the image. Depending on the dimensions, you will probably see the image move around a little in the editor. 
- Save the post and preview. 

## Screenshots 

Here are two screen captures demonstrating the use of the focal point picker:

https://cloudup.com/cIIvWAafA20

https://cloudup.com/cnL3p-eZvSq

## Types of changes

The Focal Point Picker is a new feature, which shouldn't affect anything, however the implementation in `Cover` could possibly break the block and should be tested thoroughly. Another important testing flow is to create a `Cover` block while on `master`, then branch switch to `try/focal-point-picker`. The block should remain exact as it was, with no focal point picker visible.

## Checklist:
